enterprise/server/action_cache_server_proxy/action_cache_server_proxy.go

```diff
@@ -36,19 +50,118 @@ func NewActionCacheServerProxy(env environment.Env) (*ActionCacheServerProxy, er
 	}
 	return &ActionCacheServerProxy{
 		env:         env,
+		localCache:  env.GetCache(),
 		remoteCache: remoteCache,
 	}, nil
 }
 
+func (s *ActionCacheServerProxy) getACKeyForGetActionResultRequest(req *repb.GetActionResultRequest) (*digest.ACResourceName, error) {
+	var sb strings.Builder
+	sb.WriteString(req.GetActionDigest().GetHash())
+	sb.WriteString(*actionCacheSalt)
+	if req.GetInlineStderr() {
+		sb.WriteString("|e")
+	} else {
+		sb.WriteString("|n")
+	}
+	if req.GetInlineStdout() {
+		sb.WriteString("o")
+	}
+	for _, s := range req.GetInlineOutputFiles() {
+		sb.WriteString(strconv.Itoa(len(s)))
+		sb.WriteString(s)
```

Comment:
might be also worth adding the `|` here too.

Or you could use the `hash.Strings()` util

Reply:
what would this avoid that isn't already covered by the existing code?  really i'm just trying to make it so that requests that ask for different files hash differently, which i think is already covered by this--i'd rather not compute a hash for each filename in the req

Reply:
i thiiink i convinced myself of a scenario where at least one separator was useful here, so i added it

Reply:
tyler sez: why bother.  i switched this to just hash the req